### PR TITLE
DOC: Blogpost_KD_Week (2 - 6)

### PR DESCRIPTION
**Pull request by deka27**
Summary:
This pull request introduces a blog post for week 2 - 6 of GSOC. It aims to provide insights and reflections for future work as well.

Key Changes:

Added a new blog post file in the posts/2024 directory.

### CODE REVIEW

**Review by robinroy03 — APPROVED** [review]
LGTM

**Comment by itellaetxe** [comment]
I cannot explicitly review, but LGTM

**Review by WassCodeur — APPROVED** [review]
LGTM, Great work

File: posts/2024/2024_06_12_kaustav_week_2.rst
Comment (skoudoro) [inline posts/2024/2024_06_12_kaustav_week_2.rst:16]: link to github for name

**Review by skoudoro — CHANGES_REQUESTED** [review]
Hi @deka27,

Thank you for your blogpost. For the next ones, It would be great if you could more technical and more personalized. Currently, I find the blogpost very generic. 


@WassCodeur @robinroy03 , I see LGTM but have you even try to build the blog post locally ? Do you know how to build the blog locally? Please this is a minimum. 

 I would like to see progress on reviewing code.  waiting for a second review and waiting for your feedback

File: posts/2024/2024_06_12_kaustav_week_2.rst
Comment (deka27) [reply posts/2024/2024_06_12_kaustav_week_2.rst:16]: Okay, will do this.

Comment (WassCodeur) [reply posts/2024/2024_06_12_kaustav_week_2.rst:16]: Sure

File: posts/2024/2024_06_12_kaustav_week_2.rst
Comment (WassCodeur) [inline posts/2024/2024_06_12_kaustav_week_2.rst:33]: Could you perhaps specify in which library you tested this?

File: posts/2024/2024_06_12_kaustav_week_2.rst
Comment (robinroy03) [inline posts/2024/2024_06_12_kaustav_week_2.rst:1]: `Warning, treated as error:
/home/robin/Desktop/l/dipy.org/posts/2024/2024_06_12_kaustav_week_2.rst:13:Title underline too short.`

Can be fixed by updating line-13 with an extra `-`.

**Review by robinroy03 — CHANGES_REQUESTED** [review]
Hi @deka27, please see the below changes.

Also, the order specified for a PSF blog is as follows ([source](https://social.python-gsoc.org/@terri/112638715402005382))

1. What did you do this past week?
2. What is coming up next?
3. Did you get stuck anywhere?

**Comment by skoudoro** [comment]
Also, When you update your blog post and address the comments above, Can you rebase this PR? Thank you

File: posts/2024/2024_06_12_kaustav_week_2.rst
Comment (itellaetxe) [inline posts/2024/2024_06_12_kaustav_week_2.rst:40]: I think you need to also include what are the plans for next week, and where did you get stuck, complying with the [Python GSoC Blog requirements](https://python-gsoc.org/blogs.html).
The blog only covers the first point of "what did you do this week"

You don't necessarily need to dedicate specific sections for the info that is lacking, but it should show somewhere at least.

File: posts/2024/2024_06_19_kaustav_week_3.rst
Comment (itellaetxe) [inline posts/2024/2024_06_19_kaustav_week_3.rst:41]: Same as in week 2, you need to include info about what is coming up next week, and where you got stuck.

**Review by itellaetxe — COMMENTED** [review]
Hi @deka27 sorry for the late review, I was unable to request my own review here. Now I can.

Overall, it is concise, but in my first "review comment" I did not realize about some lacking information. Could you include it, please? Just to comply with the Python GSoC blog requirements.

File: posts/2024/2024_06_12_kaustav_week_2.rst
Comment (deka27) [reply posts/2024/2024_06_12_kaustav_week_2.rst:12]: Fixed.

File: posts/2024/2024_06_12_kaustav_week_2.rst
Comment (deka27) [reply posts/2024/2024_06_12_kaustav_week_2.rst:1]: Fixed

File: posts/2024/2024_06_19_kaustav_week_3.rst
Comment (pjsjongsung) [inline posts/2024/2024_06_19_kaustav_week_3.rst:47]: The sentence here seems to be incomplete

File: posts/2024/2024_06_30_kaustav_week_4.rst
Comment (robinroy03) [inline posts/2024/2024_06_30_kaustav_week_4.rst:38]: come back stronger :zap:

File: posts/2024/2024_06_12_kaustav_week_2.rst
Comment (robinroy03) [inline posts/2024/2024_06_12_kaustav_week_2.rst:4]: The `post` date is the date when you wrote the blog. I believe you did both 2 and 3 on the same day.

**Review by robinroy03 — COMMENTED** [review]
Hi Kaustav, good work here!

The 3 blogs now answer the main [questions](https://social.python-gsoc.org/@terri/112638715402005382).
Check the below comment regarding `post` date.

File: posts/2024/2024_06_30_kaustav_week_4.rst
Comment (skoudoro) [inline posts/2024/2024_06_30_kaustav_week_4.rst:4]: Week 4 should be June 23 and week 5 is June 30

**Review by skoudoro — CHANGES_REQUESTED** [review]
Hi @deka27,

Can provide week 5, address all comments and update week 4 which has the wrong date. Thank you

File: posts/2024/2024_07_07_kaustav_week_6.rst
Comment (skoudoro) [inline posts/2024/2024_07_07_kaustav_week_6.rst:15]: why all this capital letter?

File: posts/2024/2024_07_07_kaustav_week_6.rst
Comment (skoudoro) [inline posts/2024/2024_07_07_kaustav_week_6.rst:20]: link to `lazy_loader` package

File: posts/2024/2024_07_07_kaustav_week_6.rst
Comment (skoudoro) [inline posts/2024/2024_07_07_kaustav_week_6.rst:25]: why capital to challenges?

File: posts/2024/2024_07_07_kaustav_week_6.rst
Comment (skoudoro) [inline posts/2024/2024_07_07_kaustav_week_6.rst:27]: - too many comas
- `hat`? or `that` ?
- figure  ? or `figure out` 
- `arg` to stylized

File: posts/2024/2024_07_07_kaustav_week_6.rst
Comment (skoudoro) [inline posts/2024/2024_07_07_kaustav_week_6.rst:31]: recognize instead (american style)

**Review by skoudoro — COMMENTED** [review]
Hi @deka27, see below few comments.

it would be great if @WassCodeur, @itellaetxe  and @robinroy03 could have a look also.

Thanks

File: posts/2024/2024_07_07_kaustav_week_6.rst
Comment (robinroy03) [inline posts/2024/2024_07_07_kaustav_week_6.rst:22]: Add a `\n` between challenges 1, 2 & 3.

image

**Review by robinroy03 — COMMENTED** [review]
Good work Kaustav. I'm confident you'll fix the PR challenges you've got right now.

Please see the comment below for the week 6 blog review.

File: posts/2024/2024_07_07_kaustav_week_6.rst
Comment (itellaetxe) [inline posts/2024/2024_07_07_kaustav_week_6.rst:27]: ...has issues... --> ...**it** has issues...

...related to... --> ...related to **the**...

arg --> **argument**

File: posts/2024/2024_07_07_kaustav_week_6.rst
Comment (itellaetxe) [inline posts/2024/2024_07_07_kaustav_week_6.rst:31]: ...receiving an error in importing... --> ...receiving an error **when** importing...

...as possible to scikit-image implementation... --> ...as possible to **the** `scikit-image` implementation

File: posts/2024/2024_07_07_kaustav_week_6.rst
Comment (itellaetxe) [inline posts/2024/2024_07_07_kaustav_week_6.rst:36]: Style the word 'lazy_loader' --> `lady_loader`

File: posts/2024/2024_07_07_kaustav_week_6.rst
Comment (itellaetxe) [reply posts/2024/2024_07_07_kaustav_week_6.rst:20]: Style the word 'lazy_loader' --> `lazy_loader`

**Review by itellaetxe — COMMENTED** [review]
Good work Kaustav! Looking good. Just small grammar improvements/suggestions, see below:

File: posts/2024/2024_07_07_kaustav_week_6.rst
Comment (deka27) [reply posts/2024/2024_07_07_kaustav_week_6.rst:31]: Done. Thanks.

File: posts/2024/2024_07_07_kaustav_week_6.rst
Comment (deka27) [reply posts/2024/2024_07_07_kaustav_week_6.rst:27]: Done. Thanks.

File: posts/2024/2024_07_07_kaustav_week_6.rst
Comment (deka27) [reply posts/2024/2024_07_07_kaustav_week_6.rst:15]: Done. Thanks.

File: posts/2024/2024_07_07_kaustav_week_6.rst
Comment (deka27) [reply posts/2024/2024_07_07_kaustav_week_6.rst:22]: Done. Thanks.

File: posts/2024/2024_07_07_kaustav_week_6.rst
Comment (deka27) [reply posts/2024/2024_07_07_kaustav_week_6.rst:31]: Done. Thanks.

File: posts/2024/2024_07_07_kaustav_week_6.rst
Comment (deka27) [reply posts/2024/2024_07_07_kaustav_week_6.rst:27]: Done. Thanks.
I changed arg to argument and highlighted it.

File: posts/2024/2024_07_07_kaustav_week_6.rst
Comment (deka27) [reply posts/2024/2024_07_07_kaustav_week_6.rst:25]: Fixed. Thanks.

File: posts/2024/2024_07_07_kaustav_week_6.rst
Comment (deka27) [reply posts/2024/2024_07_07_kaustav_week_6.rst:15]: Fixed. Thanks.

**Review by skoudoro — APPROVED** [review]
Ok, thanks for the update, I will not wait @WassCodeur review and go ahead and merge.

**Pull request merged**
